syncer/syncer.go

```diff
@@ -1737,6 +1737,12 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 		return nil
 	}
 
+	// eventIndex is the rows event index in this transaction, it's used to avoiding read duplicate event in gtid mode
+	eventIndex := 0
+	// the relay log file may be truncated(not end with an RotateEvent), int this situation, we may read some rows events
+	// and then read from the gtid again, so we force enter safe-mode for one more transaction to avoid failure due to
+	// conflict
+	safeMode := false
```

Comment:
I think when we meet ErrorMaybeDuplicateEvent, we know that the first half part of transaction is in the previous relay log file, so we could also use `eventIndex` to skip them.